central/debug/service/service.go

```diff
@@ -61,10 +61,11 @@ const (
 
 	centralClusterPrefix = "_central-cluster"
 
-	metricsPullTimeout     = 20 * time.Second
-	diagnosticsPullTimeout = 20 * time.Second
-	layout                 = "2006-01-02T15:04:05.000Z"
-	logWindow              = 20 * time.Minute
+	layout    = "2006-01-02T15:04:05.000Z"
+	logWindow = 20 * time.Minute
+	// This timeout is safety net to prevent request from running forever.
+	// We don't expect it to ever actually be reached. Actual timeout should be set on the client side.
+	debugDumpHardTimeout = 1 * time.Hour
```

Comment:
When it could happen?

Reply:
This is safety net against very hypothetical situation:
1. Customers use direct API call without timeout
2. Somehow pulling sensor/kubernetes data is stuck

This is basically to prevent potential resource leaking. We don't expect to run into this timeout any time soon